client/src/components/DiagnosticContainer.jsx

```diff
@@ -23,31 +24,49 @@ const dateOptions = {
   day: 'numeric',
 };
 
+// check if the survey has been submitted today
+const hasSubmitted = () => {
+  const date = window.localStorage.getItem('date');
+  const todaysDate = new Date().toISOString().slice(0, 10);
+  if (date === todaysDate) {
+    if (window.localStorage.getItem('surveyCompleted') === 'false') {
+      return false;
+    }
+    return true;
+  }
+  window.localStorage.setItem('date', todaysDate);
+  window.localStorage.setItem('surveyCompleted', 'false');
```

Comment:
Is there any benefit of using local storage here and not storing the surveyCompleted state in the Redux store?

Reply:
My thinking was that the localStorage is more persistent, so if the user opened the app in the morning, filled out the survey, then logged out, closed the app, later when the user opens the app again the same day, the survey won't be displayed after logging in. This way we can make sure that the user gets the survey as the first page only once a day

Reply:
Makes sense 👍